composer/trainer/trainer.py

```diff
@@ -580,7 +588,12 @@ def _train_loop(self) -> None:
 
                     self.engine.run_event(Event.AFTER_DATALOADER)
 
-                    microbatches = state.train_data.split_batch(state.batch, state.grad_accum)
+                    num_samples_in_batch = self.device.tensor_to_device(
+                        torch.tensor([state.batch_num_samples], dtype=torch.int))
+                    num_tokens_in_batch = self.device.tensor_to_device(
+                        torch.tensor([state.batch_num_tokens], dtype=torch.int))
+                    dist.all_reduce(num_samples_in_batch, reduce_operation="SUM")
+                    dist.all_reduce(num_tokens_in_batch, reduce_operation="SUM")
```

Comment:
Food for thought: this pattern of converting an int to a tensor, reducing the tensor, then converting back to an int recurs quite a lot. Maybe add a new helper function to `dist`? Feel free to ignore.

Reply:
I think I'll leave it for now as dist.all_reduce operates in place, which is why it requires it to be a tensor. I could see a signature like `all_reduce(input, op) -> output`, but I think that would make it unclear that `input` would be modified in place (in the case that it's a tensor).